src/Provider/JenssegersAgent.php

```diff
@@ -0,0 +1,234 @@
+<?php
+namespace UserAgentParser\Provider;
+
+use Jenssegers\Agent\Agent;
+use UserAgentParser\Exception\NoResultFoundException;
+use UserAgentParser\Exception\PackageNotLoadedException;
+use UserAgentParser\Model;
+
+class JenssegersAgent extends AbstractProvider
+{
+    /**
+     * Name of the provider
+     *
+     * @var string
+     */
+    protected $name = 'JenssegersAgent';
+
+    /**
+     * Homepage of the provider
+     *
+     * @var string
+     */
+    protected $homepage = 'https://github.com/jenssegers/agent';
+
+    /**
+     * Composer package name
+     *
+     * @var string
+     */
+    protected $packageName = 'jenssegers/agent';
+
+    protected $detectionCapabilities = [
+
+        'browser' => [
+            'name'    => true,
+            'version' => true,
+        ],
+
+        'renderingEngine' => [
+            'name'    => false,
+            'version' => false,
+        ],
+
+        'operatingSystem' => [
+            'name'    => true,
+            'version' => true,
+        ],
+
+        'device' => [
+            'model'    => true,
+            'brand'    => false,
+            'type'     => false,
+            'isMobile' => true,
+            'isTouch'  => false,
+        ],
+
+        'bot' => [
+            'isBot' => true,
+            'name'  => true,
+            'type'  => false,
+        ],
+    ];
+
+    /**
+     * Used for unitTests mocking
+     *
+     * @var Agent
+     */
+    private $parser;
+
+    /**
+     *
+     * @throws PackageNotLoadedException
+     */
+    public function __construct()
+    {
+        if (! file_exists('vendor/' . $this->getPackageName() . '/composer.json')) {
+            throw new PackageNotLoadedException('You need to install the package ' . $this->getPackageName() . ' to use this provider');
+        }
+    }
+
+    /**
+     *
+     * @return Agent
+     */
+    public function getParser()
+    {
+        if ($this->parser !== null) {
+            return $this->parser;
```

Comment:
@mimmi20 only in the unit test to mock the parser. Do you have a better idea?
https://github.com/ThaDafinser/UserAgentParser/blob/feature/agent/tests/unit/Provider/JenssegersAgentTest.php#L90-L94